api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -127,8 +146,15 @@ message CommonTlsContext {
   repeated TlsCertificate tls_certificates = 2;
 
   // Configs for fetching TLS certificates via SDS API.
-  repeated SdsSecretConfig tls_certificate_sds_secret_configs = 6
-      [(validate.rules).repeated = {max_items: 1}];
+  repeated SdsSecretConfig tls_certificate_sds_secret_configs = 6 [
+    (validate.rules).repeated = {max_items: 1},
+    (udpa.annotations.field_migrate).oneof_promotion = "dynamic_tls_certificate"
+  ];
+
+  // Config for fetching TLS certificates via CA client plugin.
```

Comment:
I think I mentioned this elsewhere, but for each of these new fields can you clarify the async loading behavior? By that I mean whether the plugin architecture supports re-loaded certs independent of the owning configuration. I believe the answer we want is "yes" because that is how SDS works, but we should be specific about that. If you could also clarify the SDS docs/comments that would be great. This comment applies to the other cases as well.

Reply:
If I understand you correctly the answer is yes: once a certificate-provider (new name for plugins) is instantiated it will keep refreshing certs (based on configured policy) and publish or return these certs for new mTLS/TLS connections created for that cluster/listener. This is similar to the SDS server which publishes up-to-date certs on the respective SDS stream. I will clarify this in the comments in this proto file and also add similar verbiage for the SDS fields. When you say "the other cases" the only other cases are file-based certs and inline certs (directly coming from xDS control plane). File-based certs can be dynamically reloaded but for inline certs there is obviously no dynamic reloading because you get a single value inline in the CDS/LDS response. Let me know if this addresses your question/concern.

Reply:
Sorry by "other cases" I just meant to add similar clarifying verbiage to all of the plugin/SDS references, not just the one I commented on. Otherwise this sounds great, thank you!